aiohttp/web_app.py

```diff
@@ -163,6 +163,7 @@ def add_subapp(self, prefix, subapp):
         self.router.register_resource(resource)
         self._reg_subapp_signals(subapp)
         self._subapps.append(subapp)
+        subapp.freeze()
```

Comment:
Even though is not part of that PR I'm wondering if makes sense does not allow add sub apps that have been frozen before. There is an explicit check for that few lines above.

Reply:
What is your use case?
Sharing sub-app instance between different parents / prefixes can open a can of worms.
The restriction was exist from very beginning IIRC

Reply:
Just trying to understand the why of this check. Looks like an implicit way to check that the sub app has been already added as child of one app (indeed could be the same).

Reply:
Yes, subapp should be used only once. If you need reuse -- please create another subapp instance.